server/auth/types/jwt/jwt_helper.test.ts

```diff
@@ -200,4 +202,153 @@ describe('test jwt auth library', () => {
 
     expect(headers).toEqual(expectedHeaders);
   });
+}); // re-import JWTAuth to change cookie splitter to a no-op
+
+/* eslint-disable no-shadow, @typescript-eslint/no-var-requires */ describe('JWT Expiry Tests', () => {
+  const setExtraAuthStorageMock = jest.fn();
+  jest.resetModules();
+  jest.doMock('../../../session/cookie_splitter', () => ({
+    setExtraAuthStorage: setExtraAuthStorageMock,
+  }));
+  const { JwtAuthentication } = require('./jwt_auth');
+
+  const realDateNow = Date.now.bind(global.Date);
+  const dateNowStub = jest.fn(() => 0);
+  global.Date.now = dateNowStub;
+  const coreSetup = jest.fn();
+
+  afterAll(() => {
+    global.Date.now = realDateNow;
+  });
+
+  test('getExpirationDate', () => {
+    expect(getExpirationDate(undefined, 1000)).toBe(1000); // undefined
+    expect(getExpirationDate('', 1000)).toBe(1000); // empty string
+    expect(getExpirationDate('Bearer ', 1000)).toBe(1000); // empty token
+    expect(getExpirationDate('Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9', 1000)).toBe(1000); // malformed token with one part
+    expect(
+      getExpirationDate(
+        'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJleGFtcGxlLmNvbSIsInN1YiI6ImpvaG4uZG9lQGV4YW1wbGUuY29tIiwiZXhwIjoxMzAwODE5MzgwMCwibmFtZSI6IkpvaG4gRG9lIiwicm9sZXMiOiJhZG1pbiJ9.ciW9WWtIaA-QJqy0flPSfMNQfGs9GEFqcNFY_LqrdII',
+        1000
+      )
+    ).toBe(1000); // JWT with very far expiry defaults to lower value (ttl)
+    expect(
+      getExpirationDate(
+        'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJleGFtcGxlLmNvbSIsInN1YiI6ImpvaG4uZG9lQGV4YW1wbGUuY29tIiwiZXhwIjo5MjA4NjkyMDAsIm5hbWUiOiJKb2huIERvZSIsInJvbGVzIjoiYWRtaW4ifQ.q8CtMfAeWOGDCGZ8UB8IIV-YM9hkDS8-pq0DSXh965I',
+        920869200001
+      )
+    ).toBe(920869200000); // JWT expiry is lower than the default
+    expect(
+      getExpirationDate(
+        'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJleGFtcGxlLmNvbSIsInN1YiI6ImpvaG4uZG9lQGV4YW1wbGUuY29tIiwibmFtZSI6IkpvaG4gRG9lIiwicm9sZXMiOiJhZG1pbiJ9.YDDoAKtA6wXd09zZ0aIUEt_IFvOwUd3rk4fW5aNppHM',
+        1000
+      )
+    ).toBe(1000); // JWT doesn't include a exp claim
```

Comment:
if I understand this correctly, the value set will be less of two properly-defined values? What happens when expiry of default is compared with a 1000? Will it set to 1000? (I think so based on the helper function but would like to confirm my understanding)

nit: this claim seems to be repeating and can be defined as a const:
```ts
const JWT_DEFAULT_EXPIRY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJleGFtcGxlLmNvbSIsInN1YiI6ImpvaG4uZG9lQGV4YW1wbGUuY29tIiwiZXhwIjo5MjA4NjkyMDAsIm5hbWUiOiJKb2huIERvZSIsInJvbGVzIjoiYWRtaW4ifQ.q8CtMfAeWOGDCGZ8UB8IIV-YM9hkDS8-pq0DSXh965I'
```
same can be applied for other tokens too.

Reply:
Basically the behavior we want to achieve for JWT is that the session is the lesser of (config's session.ttl, or the exp claim of the JWT, if it exists). Agree with your readability comment, so refactored it in: [ace02e2](https://github.com/opensearch-project/security-dashboards-plugin/pull/1773/commits/ace02e27d5b75e7a2f283bced7d7d57d516a49f3)